pandas/parser.pyx

```diff
@@ -1468,6 +1505,97 @@ cdef _string_box_decode(parser_t *parser, int col,
 
     return result, na_count
 
+@cython.boundscheck(False)
+cdef _categorical_convert(parser_t *parser, int col,
```

Comment:
what are you needing to code gen? that's not a if deal to add another function and it's at compile time